### PR TITLE
unify opencompass

### PR DESCRIPTION
**Pull request by Yunnglin**
- unify opencompass and vlmeval output path
- output path lazy initialize
- replace `|` with `union` for python 3.10 ealier

### CODE REVIEW

File: evalscope/utils/io_utils.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            raise e
+
+    return stream
+
+
+def are_paths_same(path1, path2):
+    """
+    Check if two paths are the same.
+    """
+    real_path1 = os.path.realpath(os.path.abspath(path1))
```
Comment (wangxingjun778) [inline evalscope/utils/io_utils.py]: Add `os.path.expanduser()` to expand the path like `~/.cache/path/to/xxx`

File: examples/example_eval_swift_openai_api.py [hunk 2]
```diff
@@ -47,16 +47,16 @@ def run_swift_eval():
     task_cfg = dict(
         eval_backend='OpenCompass',
         eval_config={
-            'datasets': ['winogrande'],
+            'datasets': ['gsm8k', 'ARC_c'],
             'models': [
                 {
-                    'path': 'qwen2-7b-instruct',  # Please make sure the model is deployed
+                    'path': 'qwen2.5',  # Please make sure the model is deployed
```
Comment (wangxingjun778) [inline examples/example_eval_swift_openai_api.py]: The `path` should be aligned with MS/HF model_name, e.g. for Qwen/Qwen2.5-7B-Instruct, the path should be `Qwen2.5-7B-Instruct`

File: evalscope/utils/io_utils.py (hunk 1 above)
Comment (Yunnglin) [reply evalscope/utils/io_utils.py]: done

File: examples/example_eval_swift_openai_api.py (hunk 2 above)
Comment (Yunnglin) [reply examples/example_eval_swift_openai_api.py]: done

**Pull request merged**